### PR TITLE
Adding spotiflow

### PR DESCRIPTION
**Pull request by AlbertDominguez**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by AlbertDominguez** [comment]
@conda-forge/help-python PR is ready for review!

**Comment by AlbertDominguez** [comment]
@conda-forge/help-python-c just realizing that the python-c label is more appropriate :)

File: recipes/spotiflow/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    - scikit-image
+    - scipy
+    - setuptools
+    - tensorboard
+    - tifffile
+    - torchvision
+    - tqdm
+    - typing-extensions
+    - wandb
+    - {{ pin_compatible('numpy') }}
```
Comment (xhochy) [inline recipes/spotiflow/meta.yaml:55]: Numpy has a run_exports and this should no longer be used.
```suggestion
```

File: recipes/spotiflow/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    - {{ compiler('cxx') }}
+    - make
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
+  host:
+    - python
+    - setuptools >=40.8.0
+    - setuptools-scm >=6.2
+    - wheel
+    - {{ pin_compatible('numpy') }}
```
Comment (xhochy) [inline recipes/spotiflow/meta.yaml:35]:
```suggestion
    - numpy
```

File: recipes/spotiflow/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+requirements:
+  build:
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - numpy                                  # [build_platform != target_platform]
+    - {{ compiler('c') }}
+    - {{ stdlib("c") }}
+    - {{ compiler('cxx') }}
+    - make
+    - llvm-openmp  # [osx]
```
Comment (xhochy) [inline recipes/spotiflow/meta.yaml:28]: Please also add them in host

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by AlbertDominguez** [comment]
Thanks for taking a look @xhochy! I think I've assessed all your comments now :)

File: recipes/spotiflow/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    - python
+    - setuptools >=40.8.0
+    - setuptools-scm >=6.2
+    - wheel
+    - numpy
+    - pip
+    - {{ compiler('c') }}
+    - {{ stdlib("c") }}
+    - {{ compiler('cxx') }}
+    - make
```
Comment (xhochy) [inline recipes/spotiflow/meta.yaml:40]: I only meant the OpenMP libraries. They currently have a bug where they need to be in build and host, all other should only be in build.
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by xhochy — APPROVED** [review]

**Pull request merged**